src/screens/Distribution/index.js

```diff
@@ -0,0 +1,203 @@
+import RingChart from "./RingChart";
+import { portfolioSelector } from "../../actions/portfolio";
+import CurrencyUnitValue from "../../components/CurrencyUnitValue";
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  currencies: CryptoCurrency[],
+  getCounterValue: (Currency, BigNumber) => *,
+  accounts: Account[],
+  portfolio: Portfolio,
+  counterValueCurrency: Currency,
+  t: T,
+};
+
+const mapStateToProps = createStructuredSelector({
+  getCounterValue: state => {
+    const counterValueCurrency = counterValueCurrencySelector(state);
+    const toExchange = counterValueExchangeSelector(state);
+    return (currency, value) => {
+      const currencySettings = currencySettingsSelector(state, { currency });
+      const fromExchange = currencySettings.exchange;
+      return CounterValues.calculateWithIntermediarySelector(state, {
+        from: currency,
+        fromExchange,
+        intermediary: intermediaryCurrency,
+        toExchange,
+        to: counterValueCurrency,
+        value,
+        disableRounding: true,
+      });
+    };
+  },
+  counterValueCurrency: counterValueCurrencySelector,
+  portfolio: portfolioSelector,
+  accounts: accountsSelector,
+});
+
+class Distribution extends PureComponent<Props, *> {
+  static navigationOptions = {
+    title: i18next.t("distribution.header"),
+    headerLeft: null,
+  };
+
+  renderItem = ({ item, _ }: { item: DistributionItem, index: number }) => (
+    <DistributionCard item={item} />
+  );
+
+  keyExtractor = item => item.id;
+
+  render() {
+    const distribution: AssetsDistribution = getAssetsDistribution(
+      this.props.accounts,
+      this.props.getCounterValue,
+    );
+
+    const { counterValueCurrency } = this.props;
+
+    return (
+      <SafeAreaView style={styles.wrapper}>
+        <TrackScreen category="Distribution" />
+
+        <ScrollView style={styles.root}>
+          <View style={styles.header}>
+            <View style={styles.chartWrapper}>
+              <RingChart data={distribution.list} />
+              <View style={styles.assetWrapper}>
+                <LText tertiary style={styles.assetCount}>
+                  {distribution.list.length}
+                </LText>
+                <LText tertiary style={styles.assets}>
+                  <Trans i18nKey="distribution.assets" />
+                </LText>
+              </View>
+            </View>
+            <View>
+              <LText tertiary style={styles.label}>
+                <Trans i18nKey="distribution.total" />
+              </LText>
+              <LText tertiary style={styles.amount}>
+                <CurrencyUnitValue
+                  unit={counterValueCurrency.units[0]}
+                  value={distribution.sum}
+                />
+              </LText>
+            </View>
+          </View>
+          <LText bold secondary style={styles.distributionTitle}>
+            <Trans i18nKey="distribution.list" />
+          </LText>
+          <FlatList
```

Comment:
the part above the FlatList could be put into a `HeaderComponent` so you don't need the top level ScrollView. that simplifies a bit the nesting and can be better for performance. like we did on Portfolio basically.

Reply:
Oh yeah, you mentioned this in fact